Assignments/ip1.md

```diff
@@ -0,0 +1,301 @@
+
+## Overview of Relevant Classes
+The UML class diagram below shows the three main classes that you will be implementing for this deliverable (`TicTacToeGameArea` and `TicTacToeGame`) along with several relevant classes for additional context. An `InteractableArea` represents a region of the 2D map that the player can interact with - existing implementations of this interface include `ConversationArea` and `ViewingArea`.
+
+The new `GameArea` abstraction extends `InteractableArea`, adding state to track the current `Game` in the area and the history of past games using its `GameResult` array. Subclasses of `GameArea` must implement the `handleCommand` method of `InteractableArea`, which is how messages are passed from clients to the server.
+
+The base `Game` abstract class tracks the state of the game, including who is playing it. The abstract class includes the glue that allows the socket protocol to serialize the game state such that it can be sent from the server to clients. Implementors of the `Game` abstract class implement the methods `applyMove`, `_join` and `_leave`.
+
+<script src="{{site.baseurl}}/assets/js/mermaid.min.js"></script>
+<div class="mermaid">
+%%{init: { 'theme':'forest', } }%%
+classDiagram
+    class Game {
+      +GameState state
+      +GameInstanceID id
+      ~Player[] _players
+      ~GameResult _result
+    + join(player: Player)
+    + leave(player: Player)
+    ~ _join(player: Player)
+    ~ _leave(player: Player)
+    + applyMove(move: GameMove)
+
+    }
+    class  GameArea {
+        ~Game _game
+        ~GameResult _history
+    }
+    class TicTacToeGame {
+
+    }
+    class TicTacToeGameArea {
+    }
+    class InteractableArea {
+        +string id
+       ~Player[] _occupants
+       +string[] occupantsByID
+       +boolean isActive
+       +BoundingBox boundingBox
+       +add(player: Player)
+       +remove(player: Player)
+       +addPlayersWithinBounds(allPlayers: Player[])
+       +toModel()
+       +contains(location: PlayerLocation)
+       +overlaps(otherInteractable: Interactable)
+       + handleCommand(command: InteractableCommand, player:Player)
+       #_emitAreaChanged()
+    }
+
+    class GameResult {
+        +GameInstanceID gameID
+        +Map scores
+    }
+    GameArea o-- GameResult
+    TicTacToeGame ..|> Game
+    TicTacToeGameArea ..|> GameArea
+    GameArea ..|> InteractableArea
+    GameArea o-- Game
+</div>
+
+## Implementation Tasks
+This deliverable has four parts; each part will be graded on its own rubric. You should complete the assignment one part at a time, in the order presented here.
+
+**General Requirements**: Implement your code *only* in the files specified: `src/games/TicTacToeGame.ts`, `src/games/TicTacToeGame.test.ts` and `src/games/TicTacToeGameArea.ts`. You should not install any additional dependencies. The autograder will ignore any other files that you modify, and will not install any dependencies that you add to the project.
```

Comment:
Small typo in the paths here. It should be `src/town/games/TicTacToeGame.ts`, `src/town/games/TicTacToeGame.test.ts` and `src/town/games/TicTacToeGameArea.ts`

Reply:
Are you able to push to this PR? It would be helpful if you could directly resolve these typos, thanks!